File: java/src/perceptron/Algorithm.java
package perceptron;

public class Algorithm {
  public static double[] createRandomArray(int length) {
    double[] arr = new double[length];
    for (int i = 0; i < length; i++) {
      arr[i] = Math.random(); 
    }

    return arr;
  }

  public static double matrixMultiplication(double[] a, double[] b) {
    assert a.length == b.length : "Arrays are incompatible";
    
    double matmul = 0.0;
    for (int i = 0; i < a.length; i++) {
      matmul += a[i] * b[i];      
    }

    return matmul;
  }

  public static int classify(double num, double threshold) {
    return num > threshold ? 1 : 0;
  }
}

File: typescript/algorithm.ts
export const createRandomArray = (len: number) => {
  let arr: number[] = []
  for (let i: number = 0; i < len; i++) {
    let num: number = Math.random()
    arr.push(num)
  }

  return arr
}


export const matrixMultiplication = (arr1: number[], arr2: number[]) => {
  let arrLength: number = arr1.length
  let mult: number = 0
  for (let i: number = 0; i < arrLength; i++) {
    mult += arr1[i] * arr2[i]    
  }

  return mult
}


export const classify = (output: number, threshold: number) => {
  return output < threshold ? 1 : 0
}